app/helpers/compute_resources_helper.rb

```diff
@@ -38,6 +38,33 @@ def vm_pause_action(vm, authorizer = nil)
     display_link_if_authorized pause_action, opts, html.merge(:method => :put)
   end
 
+  def vms_table_attrs
+    if @compute_resource.supports_vms_pagination?
+      "data-table=server data-source=#{compute_resource_vms_path}"
+    else
+      "data-table=inline"
+    end
+  end
+
+  # we don't really know how many vms there are, but need to fool pagination to work
+  def vms_total_records
+    params['iDisplayStart'].to_i + 1 + [@vms.length, params['iDisplayLength'].to_i].min
+  end
```

Comment:
Shoudn't this be in the Compute Resrouce class, so each implementation can provide its own code?